shopinvader_api_address_shipping_note/tests/test_shopinvader_api_address_shipping_note.py

```diff
@@ -0,0 +1,146 @@
+from odoo.addons.fastapi.tests.common import FastAPITransactionCase
+from odoo.addons.shopinvader_api_address.routers import address_router
+
+
+@tagged("post_install", "-at_install")
+class TestShopinvaderApiAddressShippingNote(FastAPITransactionCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+        extendable_registry = _extendable_registries_database.get(cls.env.cr.dbname)
+        cls.token = context.extendable_registry.set(extendable_registry)
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+
+        cls.env["res.users"].create(
+            {
+                "name": "Test User",
+                "login": "test_user",
+                "groups_id": [
+                    (
+                        6,
+                        0,
+                        [
+                            cls.env.ref(
+                                "shopinvader_api_address.shopinvader_address_user_group"
+                            ).id
+                        ],
+                    )
+                ],
+            }
+        )
+
+        cls.test_partner = cls.env["res.partner"].create(
+            {
+                "name": "FastAPI Shopinvader Address Demo",
+                "street": "rue test",
+                "zip": "1410",
+                "city": "Waterloo",
+                "country_id": cls.env.ref("base.be").id,
+            }
+        )
+
+        cls.default_fastapi_authenticated_partner = cls.test_partner
+        cls.default_fastapi_router = address_router
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        context.extendable_registry.reset(cls.token)
+        super().tearDownClass()
+
+    def test_get_shipping_address(self):
+        """
+        Test to get shipping address of authenticated_partner
+        """
+
+        with self._create_test_client(router=address_router) as test_client:
+            response: Response = test_client.get(
+                "/addresses/delivery",
+            )
+
+        self.assertEqual(
+            response.status_code,
+            status.HTTP_200_OK,
+            msg=f"error message: {response.text}",
+        )
+
+        response_json = response.json()
+        self.assertEqual(0, len(response_json))
+
+        # add shipping address
+        new_address = self.env["res.partner"].create(
+            {
+                "name": "test New Addr",
+                "street": "test Street",
+                "zip": "5000",
+                "city": "Namur",
+                "country_id": self.env.ref("base.be").id,
+                "parent_id": self.test_partner.id,
+                "type": "delivery",
+                "shipping_note": "test note",
+            }
+        )
+
+        with self._create_test_client(router=address_router) as test_client:
+            response: Response = test_client.get(
+                "/addresses/delivery",
+            )
```

Comment:
to avoid to always have to pass the router on each call to `_create_test_client` you should assign the `address_router` to the property `default_fastapi_router` into your `setUpClass`